object/slicer/slicer.go

```diff
@@ -96,64 +65,80 @@ func New(signer neofscrypto.Signer, cnr cid.ID, owner user.ID, w ObjectWriter, o
 // the produced objects. Specified session token is written to the metadata of
 // all resulting objects. In this case, the object is considered to be created
 // by a proxy on behalf of the session issuer.
-func NewSession(signer neofscrypto.Signer, cnr cid.ID, token session.Object, w ObjectWriter, opts Options) *Slicer {
+func NewSession(w ObjectWriter, token session.Object) *Slicer {
 	return &Slicer{
-		signer:       signer,
-		cnr:          cnr,
-		owner:        token.Issuer(),
 		w:            w,
-		opts:         opts,
 		sessionToken: &token,
 	}
 }
 
 // fillChildMetadata writes the metadata to all child parts of object of the same stream.
-func (x *Slicer) fillChildMetadata(obj *object.Object) {
+func (x *Slicer) fillChildMetadata(obj *object.Object, containerID cid.ID, epoch uint64, userID user.ID) {
 	currentVersion := version.Current()
 	obj.SetVersion(&currentVersion)
-	obj.SetContainerID(x.cnr)
-	obj.SetCreationEpoch(x.opts.currentNeoFSEpoch)
+	obj.SetContainerID(containerID)
+	obj.SetCreationEpoch(epoch)
+	obj.SetOwnerID(&userID)
 	obj.SetType(object.TypeRegular)
-	obj.SetOwnerID(&x.owner)
 	obj.SetSessionToken(x.sessionToken)
 }
 
-const defaultPayloadSizeLimit = 1 << 20
```

Comment:
why was it moved?

Reply:
It was just [moved](https://github.com/nspcc-dev/neofs-sdk-go/pull/466/files#diff-3b2434da11f454a062d19063e14b1de38c2f0fb6009433ee48aa21c951ea74a7R22) to file begging, where all counts should be

Reply:
dont mind moving but can it be done in a separate commit then?

Reply:
I'm not sure. I don't think someone is going to revert this commit or this change affects something

Reply:
Moved const back